custom_completions.go

```diff
@@ -476,6 +502,139 @@ func checkIfFlagCompletion(finalCmd *Command, args []string, lastArg string) (*p
 	return flag, trimmedArgs, lastArg, nil
 }
 
+// initDefaultCompletionCmd adds a default 'completion' command to c.
+// This function will do nothing if any of the following is true:
+// 1- the feature has been explicitly disabled by the program,
+// 2- c has no subcommands (to avoid creating one),
+// 3- c already has a 'completion' command provided by the program.
+func (c *Command) initDefaultCompletionCmd() {
+	if c.CompletionOptions.DisableDefaultCmd || !c.HasSubCommands() {
+		return
+	}
+
+	for _, cmd := range c.commands {
+		if cmd.Name() == compCmdName || cmd.HasAlias(compCmdName) {
+			// A completion command is already available
+			return
+		}
+	}
+
+	haveFlag := !c.CompletionOptions.DisableNoDescFlag && !c.CompletionOptions.DisableDescriptions
+
+	completionCmd := &Command{
+		Use:   compCmdName,
+		Short: "generate the autocompletion script for the specified shell",
+		Long: fmt.Sprintf(`
```

Comment:
I would argue that making this command hidden makes more sense. If I have a subcommand called `commit` I would be forced to always type more letters before tab completion works. Especially since the completion command is used once or never because the scripts are usually installed with deb/rpm.
Or you add this to the `CompletionOptions` e.g. `HideCommand`

Reply:
That's an interesting point.  Hiding the completion command to improve completion 😁

But hiding a command also removes it from the help texts. I'm concerned it would be a little too drastic.  We could definitely put an option though.  

But of the tools I know (kubectl, helm, etc) none of them choose to hide their current `completion` command, so let's leave this point open for discussion with the maintainers.